plugins/modules/system/homectl.py

```diff
@@ -0,0 +1,618 @@
+        is_active = True
+        cmd = ["systemctl", "show", "systemd-homed.service", "-p", "ActiveState"]
+        rc, show_service_stdout, stderr = self.module.run_command(cmd)
+        if rc == 0:
+            state = show_service_stdout.rsplit("=")[1]
+            if state.strip() != "active":
+                is_active = False
+        return is_active
+
+    def user_exists(self):
+        # Get user properties if they exist in json
+        exists = False
+        cmd = [self.module.get_bin_path('homectl', True)]
+        cmd.append("inspect")
+        cmd.append(self.name)
+        cmd.append("-j")
+        rc, stdout, stderr = self.module.run_command(cmd, use_unsafe_shell=True)
+        if rc == 0:
+            exists = True
+        return exists
+
+    def create_user(self):
+        record = self.create_json_record(create=True)
+        cmd = [self.module.get_bin_path('homectl', True)]
+        cmd.append("create")
+        cmd.append("--identity=-")  # Read the user record from standard input.
+        return(self.module.run_command(cmd, data=record, use_unsafe_shell=True))
+
+    def _hash_password(self, password):
+        method = crypt.METHOD_SHA512
+        salt = crypt.mksalt(method, rounds=10000)
+        pw_hash = crypt.crypt(password, salt)
+        return pw_hash
+
+    def remove_user(self):
+        cmd = [self.module.get_bin_path('homectl', True)]
+        cmd.append('remove')
+        cmd.append(self.name)
+        return(self.module.run_command(cmd, use_unsafe_shell=True))
+
+    def modify_user(self):
+        record = self.create_json_record()
+        cmd = [self.module.get_bin_path('homectl', True)]
+        cmd.append('update')
+        cmd.append(self.name)
+        cmd.append('--identity=-')  # Read the user record from standard input.
+        # Resize disksize now resize = true
+        # This is not valid in user record (json) and requires it to be passed on command.
+        if self.disksize and self.resize:
+            cmd.append('--and-resize')
+            cmd.append('true')
+        return(self.module.run_command(cmd, data=record, use_unsafe_shell=True))
+
+    def get_user_metadata(self):
+        cmd = [self.module.get_bin_path('homectl', True)]
+        cmd.append("inspect")
+        cmd.append(self.name)
+        cmd.append("-j")
+        cmd.append("--no-pager")
+        rc, stdout, stderr = self.module.run_command(cmd)
+        return stdout
+
+    # Build up dictionary to jsonify for homectl commands.
+    def create_json_record(self, create=False):
+        record = {}
+        user_metadata = {}
+        self.result['changed'] = False
+        # Get the current user record if not creating a new user record.
+        if not create:
+            user_metadata = json.loads(self.get_user_metadata())
+            # Remove elements that are not meant to be updated from record.
+            # These are always part of the record when a user exists.
+            del user_metadata['signature']
```

Comment:
Are this and the other fields always present? If not necessarily, use
```suggestion
            user_metadata.pop('signature', None)
```
instead.

Reply:
They should always be present when a user already exists as homed creates this section, but I do like this idea for safety. will add `.pop` to rest.

Reply:
done